code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -200,56 +201,45 @@
 		for(var/mob/living/L in view(1,src))
 			if(L == src || L.anchored)
 				continue
+			if(istype(L, /mob/living/simple_animal/hostile/poison/giant_spider))
+				continue
 			if(Adjacent(L))
 				choices += L
 		for(var/obj/O in src.loc)
 			if(O.anchored)
 				continue
 			if(Adjacent(O))
 				choices += O
-		cocoon_target = input(src,"What do you wish to cocoon?") in null|choices
+		var/temp_input = input(src,"What do you wish to cocoon?") in null|choices

```

Comment:
src is not really needed, plus do `as null|anything in choices` so you have a cancel button